Add docs for RocketMQ pubsub

shubham1172 opened the pull request:
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

<!--Please explain the changes you've made-->
This is a copy of the previous PR #1405 that was reverted.

## Issue reference

<!--Please reference the issue this PR will close: #[issue number]-->
Closes #2467

nyemade-uversky commented on daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md:
```diff
@@ -0,0 +1,111 @@
+
+When invoking the rocketmq pub/sub, its possible to provide an optional partition key by using the `metadata` query param in the request url.
+
+You need specifie `rocketmq-tag`,`"rocketmq-key"` in `metadata`
```
```suggestion
You need to specify `rocketmq-tag`,`"rocketmq-key"` in `metadata`
```

nyemade-uversky commented:
@berndverst would you be able to give this a quick review to make sure the spec matches the implementation of this component? 

As part of this, would you be able to confirm whether the consumer ID is set to be the application ID by default.

msfussell commented on daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md:
```diff
@@ -32,6 +32,11 @@ Table captions:
+### Alibaba Cloud
+| Name                                              | Status | Component version | Since |
+|---------------------------------------------------|--------| ---- |---------------|
+| [Alibaba Cloud RocketMQ]({{< ref setup-alicloud-rocketmq.md >}}) | Alpha | v1 | 1.8 |
```
This should not be under Alibaba. This should be under generic. Even the type says type: pubsub.rocketmq.

shubham1172 replied:
Got it. This PR was approved earlier as an Alibaba specific component https://github.com/dapr/docs/pull/1405. Moving this to a generic one.

msfussell commented on daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md:
```diff
@@ -0,0 +1,111 @@
+---
+type: docs
+title: "Alibaba Cloud RocketMQ"
```
Reword to remove all references to Alibaba. This is a generic component. As per type `type: pubsub.rocketmq`

msfussell replied:
This also means renaming this file to remove alibaba.

shubham1172 committed:
Remove Alibaba and incorporate suggestions

Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

shubham1172 committed:
Update metadata

Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

shubham1172 commented:
@msfussell, @nyemade-uversky I have updated this PR. This adds RocketMQ as a generic component. I also started digging into the actual implementation of the RocketMQ code and the original docs PR was not at all matching with the current implementation. I have updated the metadata specs and example. I also found a few issues with RocketMQ metadata which we will track with https://github.com/dapr/components-contrib/issues/1772.

msfussell commented:
@shubham1172 - Great! So this PR still needs more work on the component code before updating here. I will mark this bug as wait on Code PR.

shubham1172 commented:
@msfussell the current docs are in-line with the functionality today. Once the code PR is checked in, the docs would also need update. Can we go ahead and merge this PR to start with?, and with the next code PR we can also expect a new docs PR.

msfussell reviewed (approved):
LGTM
